### PR TITLE
Run tests in isolation when runsettings has testsettings in it.

### PR DESCRIPTION
**Pull request by Faizan2304**
Fix for issue: https://devdiv.visualstudio.com/DevDiv/VS.in%20Agile%20Testing%20IDE/_workitems/edit/505998

### CODE REVIEW

File: test/Microsoft.TestPlatform.CrossPlatEngine.UnitTests/TestEngineTests.cs [hunk 1]
```diff
@@ -199,6 +199,29 @@ public void GetDiscoveryManagerShouldNotReturnsInProcessProxyDiscoveryManagereIf
         }
 
         [TestMethod]
+        public void GetDiscoveryManagerShouldNotReturnsInProcessProxyDiscoveryManagereIfrunsettingsHasTestSettingsInIt()
+        {
+            string settingXml =
+                @"<RunSettings>
+                    <RunConfiguration>
```
Comment (smadala) [inline test/Microsoft.TestPlatform.CrossPlatEngine.UnitTests/TestEngineTests.cs]: Nit: not required Runconfiguration elements.

File: test/Microsoft.TestPlatform.CrossPlatEngine.UnitTests/TestEngineTests.cs [hunk 2]
```diff
@@ -397,6 +420,30 @@ public void GetExecutionManagerShouldNotReturnInProcessProxyexecutionManagerIfIn
             Assert.IsNotInstanceOfType(executionManager, typeof(InProcessProxyExecutionManager));
         }
 
+        [TestMethod]
+        public void GetExecutionManagerShouldNotReturnInProcessProxyexecutionManagerIfrunsettingsHasTestSettingsInIt()
+        {
```
Comment (smadala) [inline test/Microsoft.TestPlatform.CrossPlatEngine.UnitTests/TestEngineTests.cs]: Confirm negative test present.

**Review by navin22 — APPROVED** [review]

**Review by smadala — APPROVED** [review]

File: test/Microsoft.TestPlatform.CrossPlatEngine.UnitTests/TestEngineTests.cs (hunk 1 above)
Comment (Faizan2304) [reply test/Microsoft.TestPlatform.CrossPlatEngine.UnitTests/TestEngineTests.cs]: Its required

File: test/Microsoft.TestPlatform.CrossPlatEngine.UnitTests/TestEngineTests.cs (hunk 2 above)
Comment (Faizan2304) [reply test/Microsoft.TestPlatform.CrossPlatEngine.UnitTests/TestEngineTests.cs]: Yes there are negative test as well.

**Pull request merged**